### ISSUE TITLE
Error waiting for Create Service Networking Connection: Error code 9, message: Allocated IP range '172.17.0.0' not found in network.

### ISSUE THREAD

**Issue by xetra11** [hashicorp/terraform-provider-google#11100]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
Terraform v1.1.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.10.0
```

### Affected Resource(s)

* google_service_networking_connection

### Terraform Configuration Files

```tf
resource "google_compute_global_address" "default" {
  address            = "172.17.0.0"
  address_type       = "INTERNAL"
  name               = "google-managed-services-barbarus"
  network            = "https://www.googleapis.com/compute/v1/projects/barbarus/global/networks/default"
  prefix_length      = 16
  project            = var.project
  purpose            = "VPC_PEERING"
}

resource "google_service_networking_connection" "peering_connection" {
  network                 = "https://www.googleapis.com/compute/v1/projects/barbarus/global/networks/default"
  service                 = "servicenetworking.googleapis.com"
  reserved_peering_ranges = [google_compute_global_address.default.address]
}
```

### Panic Output
```
 Error: Error waiting for Create Service Networking Connection: Error code 9, message: Allocated IP range '172.17.0.0' not found in network.
│ Help Token: Ae-hA1PB7JLReDqupOdqy_il9A7QYNuLjA2AQMTFVnHD1j0W1izCtBoeCOo27VEyJd-g-QT9FA4wprKHSQoUSvCcg1YHwcGvkSlcmJk3V2riSIv_
│ 
│   with google_service_networking_connection.peering_connection,
│   on main.tf line 42, in resource "google_service_networking_connection" "peering_connection":
│   42: resource "google_service_networking_connection" "peering_connection" {
│ 
╵
```

### Expected Behavior
Should create a private connection.
### Actual Behavior
Fails when applying it although the address range was created
### Steps to Reproduce

```tf
resource "google_compute_global_address" "default" {
  address            = "172.17.0.0"
  address_type       = "INTERNAL"
  name               = "google-managed-services-barbarus"
  network            = "https://www.googleapis.com/compute/v1/projects/barbarus/global/networks/default"
  prefix_length      = 16
  project            = var.project
  purpose            = "VPC_PEERING"
}

resource "google_service_networking_connection" "peering_connection" {
  network                 = "https://www.googleapis.com/compute/v1/projects/barbarus/global/networks/default"
  service                 = "servicenetworking.googleapis.com"
  reserved_peering_ranges = [google_compute_global_address.default.address]
}

terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "4.10.0"
    }
  }
}

provider "google" {
  credentials = file("key.json")

  project = var.project
  region  = var.region
  zone    = var.zone
}
```

### Important Factoids

The address has been created successfully:

```
➜  google-cloud-sdk gcloud compute addresses list
NAME                              ADDRESS/RANGE  TYPE      PURPOSE      NETWORK  REGION  SUBNET  STATUS
google-managed-services-barbarus  172.17.0.0/16  INTERNAL  VPC_PEERING  default                  RESERVED
``` 

And the network obviously exists since it's the default one that is present when creating a new project GCP

b/308122427

**Comment by rileykarson** [comment]
Hmm- the issue appears to be in the API itself. The `google_compute_global_address` resource reports success after the API does, so the range should be available.

**Comment by DrFaust92** [comment]
Keep in mind `172.17.0.0/16` is a strange range as its sometimes reserved by google, see https://cloud.google.com/sql/docs/sqlserver/configure-private-ip#private_services_access example.

**Comment by szczeles** [comment]
@xetra11 It looks like a bug in your TF file. Replace
```
reserved_peering_ranges = [google_compute_global_address.default.address]
```
with
```
reserved_peering_ranges = [google_compute_global_address.default.name]
```
and it should be all good

**Comment by SumanthBurla** [comment]
Hi there, I'm facing the same issue. 

@szczeles, I used "**reserved_peering_ranges = [google_compute_global_address.default.name]**" still I'm getting same error. I can create IP range but can not use that to create a connection. I even tried using the console it is throwing the below error.

![image](https://user-images.githubusercontent.com/51295618/201923377-1b888331-ba8a-42eb-9c68-f24c7a969187.png)

what is the possible reason for this error? and any solution in terraform? Thanks in advance :)

**Comment by himanshuo** [comment]
[SumanthBurla](https://github.com/SumanthBurla), it looks like your issue is different than @xetra11. Can you create a new issue?

**Comment by melinath** [comment]
I was able to reproduce the reported issue and the fix describe above works for me, so I believe this issue is closed. @SumanthBurla if you are still experiencing this issue, please open a [new issue](https://github.com/hashicorp/terraform-provider-google/issues/new/choose) with more details about your configuration so that we can reproduce the behavior you're seeing. Thanks all!

**Comment by melinath** [comment]
@SumanthBurla it looks like the "Please use UpdateConnection" error you mentioned may have been fixed in 5.0 with https://github.com/GoogleCloudPlatform/magic-modules/pull/8872
